app/views/patients/_pledge.html.erb

```diff
@@ -40,19 +42,25 @@
       <% end %>
     </div>
   </div>
+  <% } %>
 <% else %>
-  <div class="row hide" data-step="1" data-title="Cancel pledge:">
-    <p class="col-sm-12">If you wish to cancel a pledge (such as to change it and resend it), please proceed to the next page.</p>
+  <div class="modal-header">
+    <h4 class="modal-title">Cancel Pledge</h4>
   </div>
-
-  <div class="row hide" data-step="2" data-title="Cancel pledge:">
-    <p class="col-sm-12">To confirm you want to cancel this pledge, please uncheck the check box below.</p>
-    <div class="col-sm-12">
-      <%= bootstrap_form_for patient, id: 'pledge-create-modal-form', remote: true do |f| %>
-        <%= f.check_box :pledge_sent, label: 'Uncheck this box to rescind the pledge' %>
-      <% end %>
+  <div class="modal-body">
+    <div class="row">
+      <div class="col-sm-12">
+        <p>Are you sure you want to cancel this pledge?</p>
+      </div>
     </div>
   </div>
+  <div class="modal-footer">
+    <button type="button" class="btn btn-default js-btn-step pull-left" style="color:red;" data-dismiss="modal">No</button>
+
+    <%= form_for @patient, html:{id:'cancel-pledge-form'}, remote: true do |f|%>    
```

Comment:
This is a little bit of a departure from what we were doing before, can you help me understand what's going on?

Reply:
Here I'm creating the footer for the cancel-pledge modal. For the submit pledge modal, we call the multi-modal partial and that generates the footer itself. The cancel pledge modal doesn't have multiple steps so we don't want to call the multi-modal partial. This means have to create a separate footer for the cancel pledge modal. Since we don't want a footer that takes us back and forth across steps, I made a footer that just has the two buttons we want without the step process. I also gave the two buttons the same classes that the multi-modal footer buttons would have to keep the same styling.

I changed the form from the bootstrap_form_for to just the form_for helper because of the way bootstrap handles dismissing modals. The normal way to close a bootstrap modal is to just throw data-dimiss:"modal" onto the button. If you do that bootstrap will call the js method preventDefault. This unintentionally stops the button from actually submitting the form. So instead I just made a button with the more basic form_for helper to handle the submission. 

Pledge.js.erb is the file that creates the modal and renders the pledge partial. I added on some javascript that lets the #cancel-form-pledge submit button close the modal as well as doing its normal job of submitting the form with the :pledge_sent attribute of false.